sw/device/lib/base/mmio.h

```diff
@@ -234,6 +234,46 @@ inline void mmio_region_nonatomic_set_mask32(mmio_region_t base,
   mmio_region_write32(base, offset, value);
 }
 
+/**
+ * Masked Register field for use in 32-bit bitfields.
+ *
+ * |index| represents a shift in bits starting form the 0 bit of a given 32-bit
+ * bitfield. It is used to zero the memory inside this bitfield by applying an
+ * inverted |mask| at the |index| offset. |value| is then capped by applying
+ * |mask| and is shifted to the |index| bits.
+ *
+ * Example, 32-bit bitfield = 0b11111111'11111111'11111111'11111111
+ *          mask = 0b11111111, index = 16, value = 0b01111110 (126)
+ *          New 32-bit bitfield = 0b11111111'01111110'11111111'11111111
```

Comment:
If you format it with triple backticks, it will stay monospaced when rendered in documentation (like markdown).
```suggestion
 * Example: 
 * ```
 * 32-bit bitfield = 0b11111111'11111111'11111111'11111111
 * set_field32( mask = 0b11111111, index = 16, value = 0b01111110 (126) )
 * result bitfield = 0b11111111'01111110'11111111'11111111
 * ```
```

Reply:
Didn't add `set_field32(...)` find it more confusing than helpful, bit done all the other changes.

Reply:
Done.